torch/_dynamo/output_graph.py

```diff
@@ -180,7 +180,11 @@ def __init__(
         super(OutputGraph, self).__init__()
         self.graph = torch.fx.Graph()
         self.graphargs: List[GraphArg] = []
-        self.tracing_context: TracingContext = TracingContext()
+        fake_mode = torch._subclasses.FakeTensorMode(
+            throw_on_data_dependent_ops=True,
+            shape_env=ShapeEnv(),
+        )
+        self.tracing_context: TracingContext = TracingContext(fake_mode)
```

Comment:
Odd that TracingContext doesn't allocate the fake mode itself

Reply:
That's not a bad idea, felt presumptuous for now though.